domain/src/application/log/LogService.ts

```diff
@@ -0,0 +1,14 @@
+import { DeviceId } from '../../domain/device/core/DeviceId.js'
+import { EnvironmentData } from '../../domain/device/core/EnvironmentData.js'
+
+export interface LogService {
+  getEnvironmentData(): Promise<EnvironmentData[]>
+
+  getDataByDeviceId(deviceId: DeviceId): Promise<EnvironmentData[]>
+
+  insertEnvironmentData(environmentData: EnvironmentData): void
+
+  updateEnvironmentData(environmentData: EnvironmentData): void
+
+  deleteEnvironmentData(id: string): void
```

Comment:
`EnvironmentData` is a value object, it doesn't make sense to delete an instance by id, because it has not id.
Moreover, `EnvironmentData` should be an immutable value (we should remove setters), and update method should not exist.
It is better instead to have a deleteEnvironmentData(sourceDevice: DeviceId, upTo: timestamp)